### PR TITLE
Custom Gamble extension

### PR DESCRIPTION
**Pull request by EllisGrovesDEV**
:3

### CODE REVIEW

File: commands/Gamba.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+                return
+            if self.supabase.table('customGamble').select('id').eq("owner", True).execute().data[0]['id'] == interaction.user.id:
+                await interaction.send("Your bet was not made, you created this gamble!", ephemeral=True)
+                return
+            newCoinBalance = existingUsersData["coins"] - betValue
+            updateUsersData = {"coins": newCoinBalance}
+            self.supabase.table('Users').update(updateUsersData).eq("id", interaction.user.id).execute().data
+            self.supabase.table('customGamble').upsert(insertGambleData).execute()
+            
+
```
Comment (EthanRowley) [inline commands/Gamba.py:35]: Extra whitespace not needed

File: commands/Gamba.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        for winner in fetchWinners:
+            userData = self.supabase.table('Users').select("id, coins").eq("id", winner['id']).execute().data[0]
+            updateData = {}
+            updateData['coins'] = int(userData['coins'])+ int((winner[self.values[0]]))*2
+            totalWon = totalWon + int((winner[self.values[0]]))
+            self.supabase.table('Users').update(updateData).eq("id", winner['id']).execute()
+        self.supabase.table('customGamble').delete().neq("id", 0).execute()
+        await interaction.message.edit(content=f"Total Money won on this bet: {totalWon}!", view=None)
+        
+
```
Comment (EthanRowley) [inline commands/Gamba.py:58]: Extra whitespace not needed

File: commands/Gamba.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        unlockButton = Button(emoji="🔐", disabled=False)
+        payoutButton = Button(label="Payout")
+        submitButton = Button(label="submit", disabled=True)
+        lockedView = View()
+        lockedView.add_item(unlockButton)
+        lockedView.add_item(payoutButton)
+        payoutView = dropdownView(self.supabase)
+        
+        
+
```
Comment (EthanRowley) [inline commands/Gamba.py]: Extra whitespace not needed

File: commands/Gamba.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            await interaction.message.edit(embed = bet_message_embed, view=lockedView)
+
+        async def payoutButton_callback(interaction):
+            if self.supabase.table('customGamble').select('id').eq("owner", True).execute().data[0]['id'] != interaction.user.id:
+                await interaction.send("You cannot payout this gamble, since you didn't create it!", ephemeral=True)
+                return
+            await interaction.message.edit(embed=bet_message_embed, view=payoutView)
+
+
+
```
Comment (EthanRowley) [inline commands/Gamba.py]: Extra whitespace not needed

File: commands/__pycache__/Gamba.cpython-311.pyc
Comment (EthanRowley) [inline commands/__pycache__/Gamba.cpython-311.pyc:1]: remove please

Comment (EthanRowley) [reply commands/__pycache__/Gamba.cpython-311.pyc:1]: remove all of them

File: main.py [hunk 5]
```diff
@@ -25,6 +25,8 @@ def getClient(self):
     "commands.cf",
     "commands.pay",
     "commands.stats",
+    "commands.help", # FIXME: This doesn't return any commands when ti runs
```
Comment (EthanRowley) [inline main.py]: Double commands.help?

File: commands/Gamba.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        ownerInitial = {'id': interaction.user.id, 'owner' : True}
+        self.supabase.table('customGamble').upsert(ownerInitial).execute()
+        
+        async def buttonCallback(interaction : Interaction):
+            modal = betInput(interaction.data["custom_id"], self.supabase)
+            await interaction.response.send_modal(modal)
+    
+        bet_message_embed = nextcord.Embed(
+            title="New Bet",
+            description=f"Bet: {bettitle}",
```
Comment (EthanRowley) [inline commands/Gamba.py:83]: betTitle?

File: commands/Gamba.py [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+    
+    
+    
+
+    async def callback(self, interaction : Interaction):
+        fetchWinners = self.supabase.table('customGamble').select(f"id, {self.values[0]}").neq(self.values[0], 0).execute().data
+        totalWon = 0
+        for winner in fetchWinners:
+            userData = self.supabase.table('Users').select("id, coins").eq("id", winner['id']).execute().data[0]
+            updateData = {}
```
Comment (EthanRowley) [inline commands/Gamba.py:51]: can you do?
updateData = {
   coins: int(userData['coins'])+ int((winner[self.values[0]]))*2
}

**Review by EthanRowley — CHANGES_REQUESTED** [review]

File: commands/Gamba.py (hunk 7 above)
Comment (EllisGrovesDEV) [reply commands/Gamba.py:51]: no actually it breaks

File: commands/Gamba.py [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+    @nextcord.slash_command(name="gamba", description="Create a custom bet!")
+    async def ex(self, 
+                 interaction : nextcord.Interaction,
+                 bettitle : str = nextcord.SlashOption(name="bet_title"),
+                 ):
+        
+        ownerInitial = {'id': interaction.user.id, 'owner' : True}
+        self.supabase.table('customGamble').upsert(ownerInitial).execute()
+        
+        async def buttonCallback(interaction : Interaction):
```
Comment (HaaZee) [inline commands/Gamba.py]: Can this method be moved to with the other callbacks

File: commands/Gamba.py [hunk 9]
```diff
@@ -0,0 +1,162 @@
+import supabase
+from typing import Optional
```
Comment (HaaZee) [inline commands/Gamba.py]: Neither of these are being used

File: commands/Gamba.py [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+            if existingUsersData["coins"] < betValue:
+                await interaction.send("Your bet was not made. Insufficient funds.", ephemeral=True)
+                return
+            if self.supabase.table('customGamble').select('id').eq("owner", True).execute().data[0]['id'] == interaction.user.id:
+                await interaction.send("Your bet was not made, you created this gamble!", ephemeral=True)
+                return
+            newCoinBalance = existingUsersData["coins"] - betValue
+            updateUsersData = {"coins": newCoinBalance}
+            self.supabase.table('Users').update(updateUsersData).eq("id", interaction.user.id).execute().data
+            self.supabase.table('customGamble').upsert(insertGambleData).execute()
```
Comment (HaaZee) [inline commands/Gamba.py]: All of this is double indented

File: commands/Gamba.py (hunk 7 above)
Comment (HaaZee) [reply commands/Gamba.py:51]: why wouldn't this work? 

updateData { "coins": int(userData['coins']) + int(winner[self.values[0]])*2 }

File: commands/Gamba.py [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+    
+    
+
+    async def callback(self, interaction : Interaction):
+        fetchWinners = self.supabase.table('customGamble').select(f"id, {self.values[0]}").neq(self.values[0], 0).execute().data
+        totalWon = 0
+        for winner in fetchWinners:
+            userData = self.supabase.table('Users').select("id, coins").eq("id", winner['id']).execute().data[0]
+            updateData = {}
+            updateData['coins'] = int(userData['coins'])+ int((winner[self.values[0]]))*2
```
Comment (HaaZee) [inline commands/Gamba.py]: Also the + missing whitespace before, and there doesn't need to be 2 sets of brackets around the second half. 

int((winner[self.values[0]]))*2 
could just be
int(winner[self.values[0]])*2

**Review by HaaZee — CHANGES_REQUESTED** [review]

**Review by HaaZee — APPROVED** [review]

**Pull request merged**